executor/executor.go

```diff
@@ -370,6 +370,12 @@ func (e *Executor) runNode(node *node.Context) {
 
 				// only after the last worker has finished should we close child channels, and we should only close them once
 				node.ShutdownOnce.Do(func() {
+					err := node.NodeProcessor.Shutdown()
```

Comment:
i'd suggest retaining the log entry before the shutdown, so that an operator can see what the last shutdown that started was just in case the shutdown operation hangs:
```log.WithField("node_id", node.Config.ID).Info("executor: shutting down node")```